src/Core/Foundation/IoC/Container.php

```diff
@@ -104,9 +105,8 @@ private function makeInstanceFromClassName($className, array $alreadySeen)
 
         if ($classConstructor) {
             foreach ($classConstructor->getParameters() as $param) {
-                $paramClass = $param->getClass();
-                if ($paramClass) {
-                    $args[] = $this->doMake($param->getClass()->getName(), $alreadySeen);
+                if ($param->getType() instanceof ReflectionNamedType && !$param->getType()->isBuiltin()) {
```

Comment:
So it means `getType()` is equivalent to `getClass()` (when the param is a class at least)